backend/Routes/journals.js

```diff
@@ -3,108 +3,264 @@ const prisma = new PrismaClient();
+                const result = await generateJournalPromptWithFeedback(analysis, feedbackData, promptHistory, apiKey);
+                const initialPrompt = result.initialPrompt;
+                const refinedPrompt = result.refinedPrompt;
+
+                // create a new journal entry with the refined prompt
+
+                const newJournal = await prisma.journal.create({
+                    data: {
+                        userEntryId: userEntry.id,
+                        prompt: initialPrompt,
+                        refinedPrompt: refinedPrompt
+                    }
+                });
+
+                // Link the new journal entry to the user entry
+                await prisma.userEntry.update({
+                    where: { id: userEntry.id },
+                    data: { journals: { connect: { id: newJournal.id } } }
+                });
+
+                return res.status(201).json({ message: 'Journal successfully created', initialPrompt: initialPrompt, refinedPrompt: refinedPrompt, journalId: newJournal.id });
+
+            }
+
+        }
+
+        // create user entry if none is found for that day
+        else {
             userEntry = await prisma.userEntry.create({
                 data: {
                     userId,
-                    date: new Date(),
+                    date: today,
                 }
-            })
+            });
+
+            //fetch user data for the past week for analysis
+
+            const entries = await fetchUserEntries(userId);
+
+            const feedbackData = await fetchFeedbackData(userId);
+
+            const promptHistory = await fetchPromptHistory(userId);
+
+            const analysis = analyzeUserData(entries);
+
+            // Generate a journal prompt using the analysis and other data
+            const apiKey = process.env.GEMINI_API_KEY;
+
+            // extract initial prompt and refined prompt from gemini
+            const result = await generateJournalPromptWithFeedback(analysis, feedbackData, promptHistory, apiKey);
+            const initialPrompt = result.initialPrompt;
+            const refinedPrompt = result.refinedPrompt;
+
+            // create a new journal entry with the refined prompt
+
+            const newJournal = await prisma.journal.create({
+                data: {
+                    userEntryId: userEntry.id,
+                    prompt: initialPrompt,
+                    refinedPrompt: refinedPrompt
+                }
+            });
+
+            // Link the new journal entry to the user entry
+            await prisma.userEntry.update({
+                where: { id: userEntry.id },
+                data: { journals: { connect: { id: newJournal.id } } }
+            });
+
+            // Return a success response with the new journal entry
+            res.status(201).json({ message: 'Journal successfully created', initialPrompt: initialPrompt, refinedPrompt: refinedPrompt, journalId: newJournal.id });
```

Comment:
Looks like this is the same line of logic as the code above. It would be nice to see that these conditional get rearranged or a helper function is added to minimize duplicated code

Reply:
Thank you for taking the time to review my code and provide feedback! I see what you mean about the duplicated code in the conditionals. I'll look into rearranging the conditionals or extracting a helper function to minimize the duplication.